doc/source/management/data_at_rest_encryption.rst

```diff
@@ -0,0 +1,213 @@
+
+To check if the required plugins are loaded you can run:
+
+.. code-block:: mysql
+
+  mysql> SELECT PLUGIN_NAME, PLUGIN_STATUS, PLUGIN_VERSION
+  FROM INFORMATION_SCHEMA.PLUGINS
+  WHERE PLUGIN_NAME LIKE 'keyring%';
+
+  +---------------+---------------+----------------+
+  | PLUGIN_NAME   | PLUGIN_STATUS | PLUGIN_VERSION |
+  +---------------+---------------+----------------+
+  | keyring_file  | ACTIVE        | 1.0            |
+  | keyring_vault | ACTIVE        | 1.0            |
+  +---------------+---------------+----------------+
+
+Usage
+-----
+
+General tablespace encryption is enabled by the following syntax extension:
+
+.. code-block:: mysql
+
+  mysql> CREATE TABLESPACE tablespace_name ... ENCRYPTION='Y'
+
+Attempts to create or to move tables, including partitioned ones, to a general
+tablespace with an incompatible encryption setting are diagnosed and aborted.
+
+As you cannot move tables between encrypted and unencrypted tablespaces,
+you will need to create another table, execute in a desired tablespace and run
+``INSERT INTO SELECT`` from the table you want to move, and then you will get
+encrypted or decrypted table with your desired content.
+
+Key Rotation
+------------
+
+Encryption key should be rotated periodically or in case it gets compromised.
+
+Similar to `InnoDB Tablespace Encryption
+<https://dev.mysql.com/doc/refman/5.7/en/innodb-tablespace-encryption.html>`_
+key can be rotated by running:
+
+.. code-block:: mysql
+
+  mysql> ALTER INSTANCE ROTATE INNODB MASTER KEY
+
+
+Example
+-------
+
+To create an encrypted tablespace run:
+
+.. code-block:: mysql
+
+  mysql> CREATE TABLESPACE foo ADD DATAFILE 'foo.ibd' ENCRYPTION='Y';
+
+To add an encrypted table to that table space run:
+
+.. code-block:: mysql
+
+   mysql> CREATE TABLE t1 (a INT, b TEXT) TABLESPACE foo ENCRYPTION="Y";
+
+Trying to add unencrypted table to this table space will result in an error:
+
+.. code-block:: mysql
+
+  mysql> CREATE TABLE t3 (a INT, b TEXT) TABLESPACE foo ENCRYPTION="N";
+  ERROR 1478 (HY000): InnoDB: Tablespace `foo` can contain only an ENCRYPTED tables.
+
+
+.. _keyring_vault_plugin:
+
+Keyring Vault plugin
+====================
+
+In |Percona Server| :rn:`5.7.20-18` a ``keyring_vault`` plugin has been
+implemented that can be used to store the encryption keys inside the
+`Hashicorp Vault server <https://www.vaultproject.io>`_.
+
+Installation
+------------
+
+The safest way to load the plugin is to do it on the server startup by
+using `--early-plugin-load variable
+<https://dev.mysql.com/doc/refman/5.7/en/server-options.html#option_mysqld_early-plugin-load>`_
+option:
+
+.. code-block:: bash
+
+  --early-plugin-load="keyring_vault=keyring_vault.so" \
+  --loose-keyring_vault_config="/home/mysql/keyring_vault.conf"
+
+It should be loaded this way to be able to facilitate recovery for encrypted
+tables. However, if you do not care about that you can also use
+``INSTALL PLUGIN`` command:
```

Comment:
I am not sure we should even mention this. --early-plugin-load is the supported way to load keyring plugins. Does upstream document INSTALL PLUGIN for keyring plugins as allowed?